lib/components/SelectionControl/ToggleSwitch/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from "react";
+import "./toggle-switch.scss";
+import clsx from "clsx";
+
+export interface SwitchToggleProps {
+    disabled?: boolean;
+    defaultChecked?: boolean;
+    onChange?: (isEnabled: boolean) => void;
+    className?: string;
+    id?: string;
+    onFocus?: () => void;
+    onBlur?: () => void;
+    checked?: boolean;
+}
+
+const ToggleSwitch = ({
+    disabled = false,
+    onChange,
+    className,
+    defaultChecked,
+    onFocus,
+    onBlur,
+    id,
+    checked,
+}: SwitchToggleProps) => {
+    const [isEnabled, setIsEnabled] = useState(defaultChecked || false);
+
+    useEffect(() => {
+        if (checked !== undefined) {
+            setIsEnabled(checked);
+        }
+    }, [checked]);
+
+    const toggleSwitch = () => {
+        if (!disabled) {
+            if (checked === undefined) {
+                const toggleEnable = !isEnabled;
+                setIsEnabled(toggleEnable);
+                onChange?.(toggleEnable);
+            } else {
+                onChange?.(!checked);
+            }
+        }
+    };
+
+    const handleKeyDown = (event: React.KeyboardEvent) => {
+        if (event.key === "Enter" || event.key === " ") {
+            event.preventDefault();
+            toggleSwitch();
+        }
+    };
+
+    return (
+        <div
```

Comment:
Can we replace this `div` with a `button`? This will help you get rid of `role="button"` and `tabIndex`, and you will be able to utilize `button`s `disabled` attribute, which will let you use `:disabled` pseudo-class in scss without the need of adding an extra `disabled` className :)
I guess `aria-` attributes will not be needed either.

Reply:
Button added, thanks